cpp/ql/test/experimental/query-tests/Security/CWE/CWE-570/semmle/tests/test.cpp

```diff
@@ -135,3 +135,74 @@ void good_new_handles_nullptr() {
   if (new (std::nothrow) int[100] == nullptr)
     return; // GOOD
 }
+
+void* operator new(std::size_t count, void*) noexcept;
+void* operator new[](std::size_t count, void*) noexcept;
+
+struct Foo {
+  Foo() noexcept;
+  Foo(int);
+
+  operator bool();
+};
+
+void bad_placement_new_with_exception_handling() {
+  char buffer[1024];
+  try { new (buffer) Foo; } // BAD
+  catch (...) {  }
+}
+
+void good_placement_new_with_exception_handling() {
+  char buffer[1024];
+  try { new (buffer) Foo(42); } // GOOD: Foo constructor might throw [FALSE POSITIVE]
```

Comment:
Can I check my understanding on this case:
 - the `new` being called is `noexcept` just like the case above, so correct error handling would be checking the return value.
 - but unlike the case above, there's a call to a user defined constructor that may throw, which may explain the `try ... except` that's been put around this code.
 - thus, though we can't see any correct handling of memory allocation failures, we lack evidence that a _wrong_ handler is present which is what this query's all about.

Reply:
Exactly right :)

Reply:
Thanks.